pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -744,11 +744,13 @@ def test_dti_cmp_tdi_tzawareness(self, other):
 
         result = dti == other
         expected = np.array([False] * 10)
-        tm.assert_numpy_array_equal(result, expected)
+        if isinstance(other, Series):
```

Comment:
so this is technically a small api change? e.g. a dti == Series will now give a boolean Series? do we do this elsewhere?

Reply:
API change, yes.  not just DTI, but any `index == series` will now return a Series[bool] instead of ndarray[bool]

Reply:
hmm then i am no sure I like this api change. We return bool ndarrays on purpose from comparisions to avoid having to align everything. I think this could potentially be a nasty user trap.

Is there a reason you want to do this (aside from reconciling / cleaning).

Reply:
My motivation is pretty much "hunt down special cases"

IIRC @jorisvandenbossche mentioned a ways back that the existing behavior is from a time before `series[bool_dtypes_series]` was supported, but i could be off on that